assets/src/edit-story/elements/text/frame.js

```diff
@@ -93,15 +94,26 @@ function TextFrame({ element: { id, content, ...rest }, wrapperRef }) {
       }
     };
 
-    const handleMouseDown = () => {
+    const handleMouseDown = (evt) => {
       clickTime = window.performance.now();
+      clickCoordinates = {
+        x: evt.clientX,
+        y: evt.clientY,
+      };
     };
 
     const handleMouseUp = (evt) => {
       const timingDifference = window.performance.now() - clickTime;
+      if (!clickCoordinates) {
+        return;
+      }
 
-      if (timingDifference > 150) {
-        // Only short clicks count.
+      const distanceMoved = Math.max(
+        Math.abs(evt.clientX - clickCoordinates.x),
+        Math.abs(evt.clientY - clickCoordinates.y)
+      );
+      if (timingDifference > 300 || distanceMoved > 1) {
```

Comment:
@dvoytenko This is now restoring the original `300` for click length and additionally checks the distance moved, allowing movement of up to 1 pixel for still entering the edit mode.

In my testing, the value when doing a click was actually always `0`, it needed some effort to do a click which moves meanwhile without the intention of dragging, however, perhaps that's also dependent on the mouse/touchpad.

LMK if you think we should set it to `0` (or assign some other value).

Reply:
@miina what you have makes a lot of sense to me. Just a couple of nits:

1. The `distanceMoved` should be manhattan distance. E.g. `abs(clientX-.x) + abs(clientY-.y)`.
2. I think threshold can be made bigger to `> 4px` manhattan.

Reply:
@dvoytenko: The allowed distance is adjusted now.